src/containers/Viewer.jsx

```diff
@@ -104,10 +105,16 @@ export class Viewer extends Component {
 }
 
 const mapStateToProps = (state, ownProps) => {
-  let set = state.photos.map(photo => photo._id)
+  let photos = []
+  if (ownProps.router.routes[1].path === 'albums') { // photos from an album
```

Comment:
I agree with you, this is a point currently affected by the Files refactoring, the router should act according to the store state and not the reverse (which is the current Photos case).